jina/optimizers/flow_runner.py

```diff
@@ -70,6 +96,8 @@ def run(
         """
 
         self._setup_workspace(workspace)
+        self._reset_callback()
```

Comment:
we discouraged using this construction right?

Reply:
it seems a smell that maybe evaluation callback should not be owned by this?

Reply:
Totally, I'll fix it. Thanks for pointing out.

Reply:
Moved the callback refreshing to optimizer. Now it makes much more sense.